crates/mysticeti-core/src/consensus/base_committer.rs

```diff
@@ -82,12 +82,21 @@ impl BaseCommitter {
         wave * wave_length + wave_length - 1 + self.options.round_offset
     }
 
+    /// Return the voting round of the specified wave. The voting round is after the leader round
+    /// of the wave.
+    fn voting_round(&self, wave: WaveNumber) -> RoundNumber {
+        self.leader_round(wave) + 1
+    }
+
     /// The leader-elect protocol is offset by `leader_offset` to ensure that different committers
     /// with different leader offsets elect different leaders for the same round number. This
-    /// function returns `None` if there are no leaders for the specified round.
+    /// function returns `None` if there are no leaders for the specified round or if the decision
+    /// round of the wave has not been reached yet.
     pub fn elect_leader(&self, round: RoundNumber) -> Option<AuthorityIndex> {
         let wave = self.wave_number(round);
-        if self.leader_round(wave) != round {
+        let decision_round = self.decision_round(wave);
+        let highest_known_round = self.block_store.highest_round();
+        if self.leader_round(wave) != round || highest_known_round < decision_round {
             return None;
```

Comment:
not sure this logic is correct. If I remember correctly the implementation of MahiMahi, we have a condition in try_commit that skips everything unless we have a quorum of blocks in the round r + wave_length. So if this condition holds, we directly look at the leader_round. 

Of course we don't have to follow that logic, but here I think that highest_know_round is not the highest round for which we have a threshold of blocks, but just the round of the hightest valid block.